bark/benchmark/scenario_dumper.py

```diff
@@ -110,3 +111,56 @@ def write_map(self, config_idx, folder):
       mapfile = params["Scenario"]["Generation"]["ConfigurableScenarioGeneration"]["MapFilename"]
       src = os.path.join("src", "database", mapfile)
       copyfile(src, os.path.join(folder, ntpath.basename(mapfile)))
+
+
+class OpenScenarioDumper(ScenarioDumper):
+  def __init__(self, base_result_folder, benchmark_result):
+    super(OpenScenarioDumper, self).__init__(
+      base_result_folder, benchmark_result)
+  
+  def GetTrajectoryPerAgent(self, world):
+    trajectory_per_agents = {}
+    for (agent_id, agent) in world.agents.items():
+      trajectory_per_agents[agent_id] = []
+      for state_action in agent.history:
+        state = state_action[0]
+        trajectory_per_agents[agent_id].append(
+          [state[0], state[1], state[2], state[3], state[4]])
+    return trajectory_per_agents
+  
+  def GetTemplates(self):
+    curr_folder = pathlib.Path(__file__).parent.absolute()
+    template_xml = str(curr_folder) + "/templates/template_traj.xml"
+    with open(template_xml, 'r') as f:
+      xml_file = etree.parse(f)
+    vertex_template = xml_file.findall('.//Vertex')[0]
+    temp_vertex = etree.tostring(vertex_template)
+    vertex_template.getparent().remove(vertex_template)
+    temp_xml = etree.tostring(xml_file)
+    return temp_vertex, temp_xml
+    
+  def write_trajectory(self, config_idx, folder):
+    histories = super().get_benchmark_result().get_history(config_idx)
+    if histories is None:
+      logging.warning("No historic state saved, cannot dump trajetory.")
+      return
+    scenario = histories[-1]  # the last state inclues all the historic states
+    world = scenario.GetWorldState()
+    trajectory_per_agents = self.GetTrajectoryPerAgent(world)
+    temp_vertex, temp_xml = self.GetTemplates()
+    
+    for agent_id, traj in trajectory_per_agents.items():
+      agent_xml = etree.fromstring(temp_xml)
+      polyline = agent_xml.findall(".//Polyline")[0]
+      for state in traj:
+        vertex = etree.fromstring(temp_vertex)
```

Comment:
maybe put this in a seperate functkion StateToVertex?

Reply:
I thought of that, but as the function is still relatively short I did not think it is necessary